packages/ogmios/src/Provider/TxSubmitProvider/OgmiosTxSubmitProvider.ts

```diff
@@ -82,4 +91,27 @@ export class OgmiosTxSubmitProvider extends RunnableModule implements TxSubmitPr
   async startImpl(): Promise<void> {
     return Promise.resolve();
   }
+
+  private async validateHandles(context: SubmitTxArgs['context']) {
+    if (context && context.handles && context.handles.length > 0) {
+      if (!this.#handleProvider) {
+        throw new Error('No handleProvider passed to submitTx, but is needed to validate handles used in transaction.');
+      }
+
+      const handleInfoList = await this.#handleProvider.resolveHandles({
+        handles: context.handles.map((handle) => handle.handle)
+      });
+
+      for (const [index, handleInfo] of handleInfoList.entries()) {
+        if (!handleInfo || handleInfo.resolvedAddresses.cardano !== context.handles[index].handle) {
```

Comment:
This looks like a bug. Please add a test that catches it before fixing.
```suggestion
        if (!handleInfo || handleInfo.resolvedAddresses.cardano !== context.handles[index].resolvedAddresses.cardano) {
```

Reply:
0dfd234d683942a8f490be4b630f098e8d0fa823 should solve this